BarrysPeanutsTemporalJava/src/main/java/barryspeanuts/ShoppingCartWorkflowImpl.java

```diff
@@ -0,0 +1,95 @@
+package barryspeanuts;
+
+import barryspeanuts.model.Purchase;
+import barryspeanuts.model.PurchaseItem;
+import barryspeanuts.task.CheckOutTaskImpl;
+import barryspeanuts.task.PayTaskImpl;
+import barryspeanuts.task.ShipTaskImpl;
+import barryspeanuts.task.WorkflowTask;
+import io.temporal.workflow.Workflow;
+import io.temporal.workflow.WorkflowQueue;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import org.slf4j.Logger;
+
+public class ShoppingCartWorkflowImpl implements ShoppingCartWorkflow {
+  private static final Logger logger = Workflow.getLogger(ShoppingCartWorkflowImpl.class);
+  ShoppingCartActivities activities = new ShoppingCartActivitiesImpl();
+  List<PurchaseItem> purchaseItems;
+  private WorkflowQueue<WorkflowTask> queue = Workflow.newWorkflowQueue(1024);
+
+  @Override
+  public void startWorkflow() {
+    logger.info("Starting Workflow for Barry's Peanuts");
+    while (true) {
+      if (Workflow.getInfo().getHistoryLength() > 2000) {
+        // do continue as new here
```

Comment:
If we're not gonna demonstrate the continue as new, we shouldn't show this part. Continue as new is not as straightforward as it seems with outstanding activities.

Can have a concept of cart completion instead of cart emptying.